File: api/check.py
from http.server import BaseHTTPRequestHandler
import json
import os


class handler(BaseHTTPRequestHandler):
    """weather.json의 updated_at 타임스탬프를 CDN 캐시 없이 직접 반환.
    폴링 시 CDN 정적 캐시를 우회하여 최신 데이터 감지를 보장한다."""

    def do_GET(self):
        json_path = os.path.join(
            os.path.dirname(os.path.abspath(__file__)), '..', 'data', 'weather.json'
        )

        try:
            with open(json_path, 'r', encoding='utf-8') as f:
                data = json.load(f)

            self._respond(200, {
                'updated_at': data.get('updated_at', ''),
                'time_display': data.get('time_display', ''),
            })
        except FileNotFoundError:
            self._respond(404, {'error': 'weather.json 파일을 찾을 수 없습니다.'})
        except Exception as e:
            self._respond(500, {'error': str(e)})

    def do_OPTIONS(self):
        self.send_response(200)
        self._cors()
        self.end_headers()

    def _respond(self, status, data):
        self.send_response(status)
        self.send_header('Content-Type', 'application/json; charset=utf-8')
        self.send_header('Cache-Control', 'no-cache, no-store, must-revalidate')
        self._cors()
        self.end_headers()
        self.wfile.write(json.dumps(data, ensure_ascii=False).encode('utf-8'))

    def _cors(self):
        origin = self.headers.get('Origin', '') if self.headers else ''
        allowed = origin if origin else '*'
        self.send_header('Access-Control-Allow-Origin', allowed)
        self.send_header('Access-Control-Allow-Methods', 'GET, OPTIONS')
        self.send_header('Access-Control-Allow-Headers', 'Content-Type')

File: functions/api/check.js
export async function onRequest(context) {
  try {
    // CDN 캐시를 완벽히 우회하기 위해 타임스탬프와 강력한 헤더를 추가하여 현재 도메인의 데이터를 가져옵니다.
    const url = new URL(context.request.url);
    const dataUrl = `${url.origin}/data/weather.json?t=${Date.now()}`;
    
    const response = await fetch(dataUrl, {
      headers: {
        'Cache-Control': 'no-store, no-cache, must-revalidate, proxy-revalidate',
        'Pragma': 'no-cache',
        'Expires': '0'
      }
    });
    
    if (!response.ok) {
      throw new Error('데이터를 불러오지 못했습니다.');
    }
    
    const data = await response.json();
    
    // 프론트엔드(script.js)가 애타게 기다리고 있는 { "updated_at": "..." } 형식으로 반환합니다.
    return new Response(JSON.stringify({ updated_at: data.updated_at }), {
      status: 200,
      headers: { 
        'Content-Type': 'application/json',
        'Cache-Control': 'no-store' // 이 API 응답 자체도 브라우저에 캐시되지 않도록 설정
      }
    });
  } catch (error) {
    return new Response(JSON.stringify({ error: error.message }), { 
      status: 500,
      headers: { 'Content-Type': 'application/json' }
    });
  }
}
